service/resource/s3objectv1/create_test.go

```diff
@@ -87,11 +89,13 @@ func Test_Resource_S3Object_newCreate(t *testing.T) {
 	resourceConfig.Clients = Clients{
 		S3: &S3ClientMock{},
 	}
-	resourceConfig.AwsService = AwsServiceMock{}
+	resourceConfig.AwsService = awsservice.AwsServiceMock{}
+	resourceConfig.CertWatcher = &CertWatcherMock{}
```

Comment:
Nice that this is mocked.

It has a different purpose but there is also the certificatetprtest package which is useful for meeting the dependencies in tests. Mentioning it in case you haven't seen it.

https://github.com/giantswarm/certificatetpr/blob/master/certificatetprtest/service.go

Reply:
Nice! Didn't know, i've removed the local mock, thx!